csm_web/frontend/src/components/MentorSection.js

```diff
@@ -0,0 +1,328 @@
+import PropTypes from "prop-types";
+import { fetchJSON, fetchWithMethod, HTTP_METHODS } from "../utils/api";
+import { SectionDetail, InfoCard, SectionSpacetime } from "./Section";
+import { Switch, Route } from "react-router-dom";
+import { groupBy } from "lodash";
+import CopyIcon from "../../static/frontend/img/copy.svg";
+import CheckCircle from "../../static/frontend/img/check_circle.svg";
+import { ATTENDANCE_LABELS } from "./Section";
+export default function MentorSection({ id, url, course, courseTitle, spacetime, override }) {
+  const [{ students, attendances, loaded }, setState] = useState({ students: [], attendances: {}, loaded: false });
+  useEffect(() => {
+    setState({ students: [], attendances: {}, loaded: false });
+    fetchJSON(`/sections/${id}/students/`).then(data => {
+      const students = data.map(({ name, email, id }) => ({ name, email, id }));
+      const attendances = groupBy(
+        data
+          .flatMap(({ name, id, attendances }) =>
+            attendances.map(attendance => ({ ...attendance, student: { name, id } }))
+          )
+          .reverse(),
+        attendance => attendance.weekStart
+      );
+      setState({ students, attendances, loaded: true });
+    });
+  }, [id]);
+
+  const updateAttendance = (updatedWeek, updatedWeekAttendances) => {
+    const updatedAttendances = Object.fromEntries(
+      Object.entries(attendances).map(([weekStart, weekAttendances]) => [
+        weekStart,
+        weekStart == updatedWeek ? [...updatedWeekAttendances] : weekAttendances
+      ])
+    );
+    setState({ students, loaded, attendances: updatedAttendances });
+  };
+
+  return (
+    <SectionDetail
+      course={course}
+      courseTitle={courseTitle}
+      isStudent={false}
+      links={[
+        ["Section", url],
+        ["Attendance", `${url}/attendance`],
+        ["Roster", `${url}/roster`]
+      ]}
+    >
+      <Switch>
+        <Route
+          path={`${url}/attendance`}
+          render={() => (
+            <MentorSectionAttendance attendances={attendances} loaded={loaded} updateAttendance={updateAttendance} />
+          )}
+        />
+        <Route path={`${url}/roster`} render={() => <MentorSectionRoster students={students} loaded={loaded} />} />
+        <Route
+          path={url}
+          render={() => (
+            <MentorSectionInfo students={students} loaded={loaded} spacetime={spacetime} override={override} />
+          )}
+        />
+      </Switch>
+    </SectionDetail>
+  );
+}
+
+MentorSection.propTypes = {
+  id: PropTypes.number.isRequired,
+  course: PropTypes.string.isRequired,
+  courseTitle: PropTypes.string.isRequired,
+  spacetime: PropTypes.object.isRequired,
+  override: PropTypes.object,
+  url: PropTypes.string.isRequired
+};
+
+const MONTH_NUMBERS = Object.freeze({
+  Jan: 1,
+  Feb: 2,
+  Mar: 3,
+  Apr: 4,
+  May: 5,
+  Jun: 6,
+  Jul: 7,
+  Aug: 8,
+  Sep: 9,
+  Oct: 10,
+  Nov: 11,
+  Dec: 12
+});
+
+function formatDate(dateString) {
```

Comment:
Where are these date strings being sourced from? Is it possible to find a way to get this with [Date.toLocaleDateString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString) or similar?

Reply:
They come from our backend's use of Python strftime. Run the following in a Python interpreter for confirmation
```python
from datetime import date
[date(year=2020, month=i, day=1).strftime("%b") for i in range(1,13)]
```

I suggest we leave them hardcoded because they are tied to the backend, not to any source of truth within the JavaScript standard.